btcjson/cmdparse.go

```diff
@@ -475,7 +475,9 @@ func assignField(paramNum int, fieldName string, dest reflect.Value, src reflect
 					paramNum, fieldName, destBaseType)
 				return makeError(ErrInvalidType, str)
 			}
-			dest.Set(reflect.ValueOf(concreteVal).Elem())
+			if concreteVal != nil {
+				dest.Set(reflect.ValueOf(concreteVal).Elem())
```

Comment:
Would this ever not be the case because if it were nil, then its Kind would not be Array, Slice, Struct, or Map? I'm curious about the reason for this condition.